backend/kernelCI_app/views/treeCommitsHistory.py

```diff
@@ -1,155 +1,241 @@
+from kernelCI_app.utils import (
+    FilterParams,
+    InvalidComparisonOP,
+    getErrorResponseBody
+)
 from rest_framework.views import APIView
 from rest_framework.response import Response
 from rest_framework import status
 from django.http import JsonResponse, HttpResponseBadRequest
 from django.db import connection
-from kernelCI_app.utils import (
-    NULL_STRINGS,
-    FilterParams,
-    InvalidComparisonOP,
-    getErrorResponseBody,
+from typing import Dict, List
+from kernelCI_app.helpers.filters import (
+    UNKNOWN_STRING
 )
 
 
 # TODO Move this endpoint to a function so it doesn't
 # have to be another request, it can be called from the tree details endpoint
 class TreeCommitsHistory(APIView):
     def __init__(self):
-        self.filters_options = {
-            'build': {
-                'table_alias': 'b',
-                'filters': []
+        self.commit_hashes = {}
+        self.filterParams = None
+        self.field_values = dict()
+        self.processed_builds = set()
+        self.processed_tests = set()
+
+    def setup_filters(self):
+        self.filterTestDurationMin = self.filterParams.filterTestDurationMin
+        self.filterTestDurationMax = self.filterParams.filterTestDurationMax
+        self.filterBootDurationMin = self.filterParams.filterBootDurationMin
+        self.filterBootDurationMax = self.filterParams.filterBootDurationMax
+        self.filterBuildDurationMin = self.filterParams.filterBuildDurationMin
+        self.filterBuildDurationMax = self.filterParams.filterBuildDurationMax
+        self.filterBootStatus = self.filterParams.filterBootStatus
+        self.filterTestStatus = self.filterParams.filterTestStatus
+        self.filterTreeDetailsConfigs = self.filterParams.filterConfigs
+        self.filterTreeDetailsCompiler = self.filterParams.filterCompiler
+        self.filterArchitecture = self.filterParams.filterArchitecture
+        self.filterHardware = self.filterParams.filterHardware
+        self.filterTestPath = self.filterParams.filterTestPath
+        self.filterBootPath = self.filterParams.filterBootPath
+        self.filterBuildValid = self.filterParams.filterBuildValid
```

Comment:
The issueFilter is still not being applied to the commitHistoryGraph, I think it can be done in this PR as well, can't it?